tests/test_metrics.py

```diff
@@ -117,8 +116,102 @@ def test_calculate(self):
         nx_reciprocity = nx.algorithms.reciprocity(G) / 2  # nx counts each reciprocity twice
         self.assertEqual(reciprocty_fraction, nx_reciprocity)
 
+class TestDegreeMetrics(unittest.TestCase):
+    def test_get_effective_feature_count(self):
+        receiver = InDegree()
+        n = 18
+        self.assertEqual(receiver.get_effective_feature_count(n), n)
+
+        receiver = InDegree(base_idx=1)
+        n = 18
+        self.assertEqual(receiver.get_effective_feature_count(n), n-1)
+
+        sender = OutDegree()
+        n = 18
+        self.assertEqual(sender.get_effective_feature_count(n), n)
+
+        sender = OutDegree(base_idx=1)
+        n = 18
+        self.assertEqual(sender.get_effective_feature_count(n), n-1)
+    
+    def test_calculate(self):
+        W = np.array([
+            [0, 1, 0, 1],
+            [1, 0, 0, 1],
+            [1, 1, 0, 1],
+            [1, 0, 0, 0]
+        ])
+
+        G = nx.from_numpy_array(W, create_using=nx.DiGraph)
+        expected_in_degrees = list(dict(G.in_degree()).values())
+
+        receiver = InDegree()
+        indegree = receiver.calculate(W)
+
+        self.assertTrue(np.all(indegree == expected_in_degrees))
+
+        expected_out_degrees = list(dict(G.out_degree()).values())
+
+        sender = OutDegree()
+        outdegree = sender.calculate(W)
+
+        self.assertTrue(np.all(outdegree == expected_out_degrees))
+
 
 class TestMetricsCollection(unittest.TestCase):
+    def test_metrics_setup(self):
+        metrics = [NumberOfEdgesDirected(), TotalReciprocity(), InDegree()]
+        collection = MetricsCollection(metrics, is_directed=True)
+        
+        expected_metric_names = tuple([str(NumberOfEdgesDirected()), str(TotalReciprocity()), str(InDegree())])
+
+        self.assertTrue(np.all(collection.metric_names == expected_metric_names))
```

Comment:
Why `np.all` and not python's default `all`? What numpy does when it doesn't get numpy arrays?